### PR TITLE
Negate axis offset and then turn it into a string

### PR DESCRIPTION
**Pull request by etpinard**
fixes https://github.com/plotly/plotly.js/issues/2338

Not 100% sure how to test this. There's probably a way to add a check in `d3-strict`? But maybe just a jasmine test that doesn't fail would do the trick.

cc @alexcjohnson

### CODE REVIEW

File: src/plots/ternary/ternary.js [hunk 1]
```diff
@@ -328,12 +328,12 @@ proto.adjustLayout = function(ternaryLayout, graphSize) {
     _this.layers.bgrid.attr('transform', bTransform);
 
     var aTransform = 'translate(' + (x0 + w / 2) + ',' + y0 +
-        ')rotate(30)translate(0,-' + aaxis._offset + ')';
+        ')rotate(30)translate(0,' + -aaxis._offset + ')';
```
Comment (etpinard) [inline src/plots/ternary/ternary.js]: When `_offset` is negative we use to get:

image

that is double `-` in the translate transform string.

Looks like `nw.js@0.12` doesn't care, so this went unnoticed in our image tests.

**Comment by alexcjohnson** [comment]
Great, was hoping it would be that simple - does this fix the hover labels too?

A `d3-strict` test could work - if setting `transform` check that `--` doesn't appear in the string?

**Comment by etpinard** [comment]
> does this fix the hover labels too?

This one is bit tricky, `hovermode` defaults to `'x'` in `plot_types.json`, which isn't compatible with subplot types other than `cartesian`. With `hovermode: 'closest'`, everything works fine. Now, gl3d and pie seem not to care when `hovermode: 'x'` is set (they behave the same as for `hovemode: 'closest'`) whereas geo, ternary and polar `hovermode: 'x'` show an odd-looking arrow at the bottom of the corresponding subplots.

Clearly those odd-looking arrow need to go. But I ask: should `hovermode: 'x'` for subplot types without and x-axis behave like `hovermode: 'closest'` or like `hovermode: false`?

**Comment by alexcjohnson** [comment]
> should `hovermode: 'x'` for subplot types without and x-axis behave like `hovermode: 'closest'` or like `hovermode: false`?

I think anything that's not explicitly `hovermode: false` should behave like `'closest'` on subplots where compare mode doesn't make sense.

**Comment by etpinard** [comment]
> I think anything that's not explicitly hovermode: false should behave like 'closest' on subplots where compare mode doesn't make sense.

Sounds good. Let me fix that in this PR.

**Comment by etpinard** [comment]
@alexcjohnson if you're looking to fix an easy (I think) bug before the `1.34.0` release, you can fixing the `hovermode` part of https://github.com/plotly/plotly.js/issues/2338 as described in https://github.com/plotly/plotly.js/pull/2339#issuecomment-363531321 . Feel free to push to this branch directly.

I have to keep working on the `mapbox-gl` update until Wednesday.

File: src/components/fx/hover.js [hunk 2]
```diff
@@ -203,6 +205,8 @@ function _hover(gd, evt, subplot, noHoverEvent) {
 
     var hovermode = evt.hovermode || fullLayout.hovermode;
 
+    if(hovermode && !supportsCompare) hovermode = 'closest';
```
Comment (alexcjohnson) [inline src/components/fx/hover.js]: The second part of #2338 - fixes hover labels for geo and ternary on the `plot_types` mock

Comment (etpinard) [reply src/components/fx/hover.js]: Looks good. Not sure what's the best to :lock: this down?

Maybe just checking that `hovermode: 'x'` behaves the same as `hovermode: 'closest'` in a cartesian + geo (or ternary or polar) graph?

File: src/components/modebar/manage.js [hunk 3]
```diff
@@ -100,55 +103,71 @@ function getButtonGroups(gd, buttonsToRemove, buttonsToAdd) {
...
-        addGroup(['hoverClosestPie']);
-    } else if(hasMapbox) {
-        addGroup(['resetViewMapbox', 'toggleHover']);
-    } else if(hasGeo) {
-        addGroup(['zoomInGeo', 'zoomOutGeo', 'resetGeo']);
-        addGroup(['hoverClosestGeo']);
-    }
+    addGroup(dragModeGroup);
+    addGroup(zoomGroup.concat(resetGroup));
+    addGroup(hoverGroup);
```
Comment (etpinard) [inline src/components/modebar/manage.js]: Nice cleanup. Looking good!

File: src/components/fx/hover.js (hunk 2 above)
Comment (alexcjohnson) [reply src/components/fx/hover.js]: good call -> https://github.com/plotly/plotly.js/pull/2339/commits/16b01f002b4b2874ff191bde6ce9cc9f9f38f1db

**Comment by etpinard** [comment]
Nicely done :dancer:

**Comment by alexcjohnson** [comment]
Likewise, 💃 on the `--` fix and `d3-strict` test 🎉

**Pull request merged**